### PR TITLE
Add CocoaPods downloads badges.

### PR DESCRIPTION
**Pull request by nlutsenko**
Expands CocoaPods integration using `metrics.cocoapods.org`.
Tried my best to match the existing style for other package managers.

Also cc @orta @segiddins @kylef as core CocoaPods people.

### CODE REVIEW

**Comment by nlutsenko** [comment]
Also, tested locally and this is the output :grin:
screen shot 2016-05-04 at 7 37 03 pm

**Comment by segiddins** [comment]
@nlutsenko perhaps these should use apps rather than download counts?

**Comment by nlutsenko** [comment]
I think we should do both, tbh, but not sure where to put 2 more metrics for `apps` and what would the badge look like?

**Comment by nlutsenko** [comment]
@espadrine, does this look reasonable to you? Would love to have this merged!

**Comment by orta** [comment]
I'd also love to see the apps in there, FWIW, far more interesting metadata 💃 (though usually significantly smaller number :D )

**Comment by nlutsenko** [comment]
Very well, then. Let me add `apps` here as well. :D
Update to the PR is incoming.

**Comment by nlutsenko** [comment]
Perfect, this is updated now and has `apps` badges, per popular request.
And, of course, here is a fancy and visually appealing screenshot 
screen shot 2016-05-11 at 9 55 48 pm

This is also rebased on the latest master, cleaned, rinsed and ready for review.

**Comment by orta** [comment]
👍 thanks @nlutsenko

**Comment by nlutsenko** [comment]
@espadrine, if I rebase it - would you want to merge this one in?
We have a buy-in from the CocoaPods core team, and it's a pretty useful addition to the shields list, and can increase everyone's ego growth by some percentage (or at least I very much hope so).

File: server.js [hunk 1]
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
...
 }));
 
+// Cocoapods Downloads integration.
+camp.route(/^\/cocoapods\/(dm|dw|dt)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var info = match[1]; // One of these: "dm", "dw", "dt"
+  var spec = match[2];  // eg, AFNetworking
+  var format = match[3];
+  var apiUrl = 'http://metrics.cocoapods.org/api/v1/pods/' + spec;
+  var badgeData = getBadgeData('pod', data);
```
Comment (espadrine) [inline server.js]: 'downloads' instead of 'pod'. It incidentally removes the need for `getLabel()` below.

File: server.js [hunk 2]
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
...
+}));
+
+// CocoaPods Apps Integration
+camp.route(/^\/cocoapods\/(aw|at)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var info = match[1]; // One of these: "aw", "at"
+  var spec = match[2];  // eg, AFNetworking
+  var format = match[3];
+  var apiUrl = 'http://metrics.cocoapods.org/api/v1/pods/' + spec;
+  var badgeData = getBadgeData('pod', data);
```
Comment (espadrine) [inline server.js]: 'apps' instead of 'pod', which also removes the need for `getLabel()` below.

File: server.js [hunk 3]
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
...
+  request(apiUrl, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var data = JSON.parse(buffer);
+      badgeData.text[0] = getLabel('apps', data);
+      var apps = 0
```
Comment (espadrine) [inline server.js]: Missing semicolon.

File: server.js [hunk 4]
```diff
@@ -2719,6 +2719,84 @@ cache(function(data, match, sendBadge, request) {
...
+      switch (info.charAt(1)) {
+        case 'w':
+          var apps = data.stats.app_week;
```
Comment (espadrine) [inline server.js]: No need to define `apps` with var, as it was already defined in the function.

**Comment by espadrine** [comment]
Thanks a lot for the contribution! I applied the adjustments from my review myself, as they are rather small.

It is live, but I'll update the website in 24h. You can check that everything works fine in the meantime.

**Pull request merged**